crates/pevm/tests/common/mod.rs

```diff
@@ -55,14 +56,9 @@ pub fn for_each_block_from_disk(mut handler: impl FnMut(Block, InMemoryStorage<'
         )
         .unwrap();
 
-        // Parse block hashes
-        let block_hashes: BlockHashes = File::open(block_dir.join("block_hashes.json"))
-            .map(|file| serde_json::from_reader::<_, BlockHashes>(BufReader::new(file)).unwrap())
-            .unwrap_or_default();
-
         handler(
             block,
-            InMemoryStorage::new(accounts, Some(&bytecodes), block_hashes),
+            InMemoryStorage::new(accounts, Some(&bytecodes), block_hashes.clone()),
```

Comment:
Maybe the data amount is not much, but this is for the sake of consistency.

Also, `.clone` is widely considered as anti-pattern when using reference (`&`) is easy.

Reply:
Just in case, `InMemoryStorage` requires iterator with owned values, so its either `block_hashes.iter().cloned()` or just `block_hashes.clone()`.

https://github.com/risechain/pevm/blob/09bb59ad39f5df4d092b8401520b4e8e25d691d1/crates/pevm/src/storage/in_memory.rs#L21

Reply:
This means `InMemoryStorage` should be changed to accept a reference.

See https://github.com/risechain/pevm/pull/245/files#diff-d83915c2d163041919e1ac073fb26aaa65dcc05ebb7ceeaab1b70033ab85d5e8L15

Reply:
The spirit is the same. `bytecodes` and `block_hashes` are readonly values. `InMemoryStorage` only "views" them, not own them.

Reply:
I believe a reference is sufficient. Using `Arc` will be an overkill. Anyway, it's not bad.

`Arc` comes with a small performance cost. Therefore, please run a benchmark (using `cargo bench` and `crates/pevm/benches/report.py`) to make sure that the performance is not affectted.

Reply:
It shouldn't though. It is just a fancy pointer to `(AtomicUsize, AtomicUsize, T)`, so, each time you dereference `Arc` you just read pointer (but now it is with offset of 2 machine words) :man_shrugging:

Reply:
But my take here is that it would be easier to manage `Arc` over wrangling (now 2) lifetimes.

Reply:
I'm not a fan of premature `Arc`s over `&` but 2 `Option<&>`s with 2 potentially different lifetimes look pretty unpleasant. So yeah, `Arc` is justified here.